client/components/tinymce/plugins/mentions/mentions.jsx

```diff
@@ -0,0 +1,218 @@
+			const isLineAfter = ( this.top < currentPosition.top ) && ( this.left > currentPosition.left );
+
+			if ( isLineBefore || isLineAfter ) {
+				this.updatePosition( currentPosition );
+			}
+		}
+	}
+
+	componentWillUnmount() {
+		const { editor } = this.props;
+
+		editor.off( 'keyup', this.onKeyUp );
+		editor.off( 'click', this.hidePopover );
+	}
+
+	getPosition() {
+		let left, top;
+		const { editor } = this.props;
+		const nodePosition = tinymce.$( editor.selection.getNode() ).offset();
+		const rectList = editor.selection.getRng().getClientRects();
+
+		if ( rectList.length > 0 ) {
+			left = rectList[ 0 ].left;
+			top = rectList[ 0 ].top + rectList[ 0 ].height;
+		} else {
+			left = nodePosition.left;
+			top = nodePosition.top;
+		}
+
+		return { left: left, top: top };
+	}
+
+	updatePosition( position = this.getPosition() ) {
+		const { editor, node } = this.props;
+		const selectedNode = editor.selection.getNode();
+		const nodePosition = tinymce.$( selectedNode ).offset();
+		const mceToolbar = tinymce.$( '.mce-toolbar-grp', editor.getContainer() )[ 0 ];
+
+		this.left = position.left;
+		this.top = position.top;
+		node.style.left = this.left + 'px';
+		node.style.top = nodePosition.top + mceToolbar.offsetHeight + selectedNode.offsetHeight + 'px';
+	}
+
+	setPopoverContext = ( popoverContext ) => {
+		if ( popoverContext ) {
+			this.setState( { popoverContext } );
+		}
+	}
+
+	getQueryText() {
+		const range = this.props.editor.selection.getRng();
+		const textBeforeCaret = range.startContainer.textContent.slice( 0, range.startOffset );
+		const matcher = new RegExp( '(?:^|\\s)@([A-Za-z0-9_\+\-]*)$|(?:^|\\s)@([^\\x00-\\xff]*)$', 'gi' );
+		const match = matcher.exec( textBeforeCaret );
+
+		if ( match ) {
+			return match[ 2 ] || match[ 1 ];
+		}
+
+		return null;
+	}
+
+	onKeyUp = ( { keyCode } ) => {
+		if ( includes( [ VK.ENTER, VK.SPACEBAR, VK.UP, VK.DOWN, 27 /* ESCAPE */ ], keyCode ) ) {
+			return this.hidePopover();
+		}
+
+		const query = this.getQueryText();
+
+		this.setState( {
+			showPopover: query !== null,
+			query,
+		} );
+	}
+
+	handleClick = ( suggestion ) => {
+		const { editor } = this.props;
+		const re = /@\S*/;
+		const markup = <EditorMention username={ suggestion.user_login } />;
+		const range = editor.selection.getRng();
+		const mentionRange = document.createRange();
+
+		// Set a range for the user-entered mention.
+		mentionRange.setStart( range.startContainer, Math.max( range.startOffset - ( this.state.query.length + 1 ), 0 ) );
+		mentionRange.setEnd( range.endContainer, range.endOffset );
+		editor.selection.setRng( mentionRange );
+
+		// Replace user-entered mention with full username.
+		editor.selection.setContent( editor.selection.getContent().replace( re, ReactDomServer.renderToStaticMarkup( markup ) ) );
+
+		editor.getBody().focus();
+	}
+
+	hidePopover = () => {
+		this.setState( { showPopover: false } );
+	}
```

Comment:
no need for the curlies `{}` here :smile:

```js
hidePopover = () => this.setState( { showPopover: false } );
```

hmm…I wonder if I have already left this comment elsewhere. if so, I apologize. I think there are two PRs for this I'm confusing